lib/temporal.js

```diff
@@ -263,10 +278,32 @@ exportable.repeat = function(n, ms, callback) {
   });
 };
 
-exportable.clear = function() {
-  isProcessing = false;
-  exportable.removeAllListeners();
-  queue = {};
+exportable.clear = function(id) {
+
+  if (typeof id === "undefined") {
+    isProcessing = false;
+    queue = {};
+  } else {
+    queue = filterQueue(queue, id);
+  }
+
+  if (Object.keys(queue).length === 0) {
+    exportable.removeAllListeners();
+  }
+
 };
 
+function filterQueue(queue, id) {
+  var filtered = {};
+  Object.keys(queue).forEach(function(key) {
+    queue[key] = queue[key].filter(function(entry) {
+      return entry.id !== id;
+    });
+    if (queue[key].length > 0) {
+      filtered[key] = queue[key];
+    }
+  });
+  return filtered;
+}
```

Comment:
This could be potentially expensive, but the optimization I'm thinking of is non-trivial. I'm going to experiment with your branch and see if it can work the way I'm thinking and I will get back to you

Reply:
My idea is to just mark `entry.isRunnable = false` for all tasks with a given `id`.